feat: footnotes

tycebrown opened the pull request:
<!--
  Thank you for your PR! Please follow this template to help us review your PR quickly.
-->

## What has changed
- Added footnotes section to Notes tab
- Created `Footnote` schema
- Rename paths: 
  - `/languages/{code}/verses/{verseId}/notes` --> `/languages/{code}/verses/{verseId}/translator-notes`
  - `/languages/{code}/words/{wordId}/note` --> `/languages/{code}/words/{wordId}/note`
- Created paths:
  - GET: `/languages/{code}/verses/{verseId}/footnotes` 
  - PATCH: `/languages/{code}/words/{wordId}/footnote`

## Connected Issues
closes #311
<!--
  If this PR resolves an issue, please add `closes #issue-no` below to connect the issue to the PR.
-->

## QA Steps

<!-- Please describe how to test what you've changed. -->
1. Login as Admin.
1. Open the sidebar for a word and go to the Notes tab.
2. Confirm that there is a "Footnotes" section in the panel with rich text editor.
3. Edit the footnotes.
4. Click outside the editor. Confirm that the footnote's content was saved and the timestamp was updated.
5. Switch to a different word and switch back to the original word. Confirm that the previous changes persisted.
6. Reload the page. Again, confirm that the previous changes persisted.

## Post-Deployment

<!--
  Please describe any tasks that must be executed after this change is deployed.
-->

- [ ] Nothing required

tycebrown committed:
Update pre-commit

tycebrown committed:
chore; update gitignore to ignore nx cache

tycebrown committed:
Merge remote-tracking branch 'upstream/main'

tycebrown committed:
Merge remote-tracking branch 'upstream/main'

tycebrown committed:
Merge remote-tracking branch 'upstream/main'

tycebrown committed:
Merge remote-tracking branch 'upstream/main'

tycebrown committed:
Merge remote-tracking branch 'upstream/main'

tycebrown committed:
preparations

tycebrown committed:
ui setup

tycebrown committed:
db and api

tycebrown committed:
Merge remote-tracking branch 'upstream/main' into tyce-2-311-footnotes

tycebrown committed:
fix: footnote not updating

tycebrown committed:
fix: notes initially show empty

tycebrown committed:
fix: internationalization

tycebrown committed:
add a little spacing

tycebrown committed:
refactor: styling and rename

tycebrown committed:
api docs

tycebrown committed:
final cleaning

tycebrown committed:
spelling & formatting correction

arrocke commented on packages/db/src/schema.prisma:
```diff
@@ -213,8 +217,10 @@ model Language {
   /// The list of machine glosses of different words for this language.
   machineGlosses      MachineGloss[]
-  // The list of translator notes of different words for this language.
+  // The list of footnotes of different words for this language.
```
This comment doesn't need to be changed

tycebrown replied:
Done.

arrocke commented on packages/api/pages/api/languages/[code]/verses/[verseId]/footnotes.ts:
```diff
@@ -0,0 +1,47 @@
+import { client } from '../../../../../../shared/db';
+
+export default createRoute<{ code: string; verseId: string }>()
+  .get<void, GetVerseFootnotesResponseBody>({
```
We are making a ton of requests to the server on the translation view, so I think we should consolidate this and the translator's note route. It can be be renamed to end in `/notes` and have to top level keys: `footnotes` and `translatorNotes`

tycebrown replied:
Done. @arrocke Do you think we should also consolidate the `/note` and `/footnote` paths?

arrocke commented on packages/web/src/features/translation/TranslationSidebar.tsx:
```diff
@@ -148,40 +187,71 @@ export const TranslationSidebar = ({
+                        <RichText content={translatorNoteContent} />
+                      )}
+                    </div>
+                    <div className="flex flex-col gap-2">
+                      <h2 className="font-bold">{t('translate:footnotes')}</h2>
+                      {footnote?.authorName && (
+                        <span className="italic">
+                          {t('translate:note_description', {
```
Everyone can view footnotes, so we will want to conditionally render these differently. We don't need to check that they are a language member before showing them the note. That's the difference between these and translator's notes

tycebrown replied:
Ah, makes sense 👍

tycebrown replied:
Done. Also: now, only the language members and platform admin can see the note description.

arrocke reviewed (changes requested):
Nice work on this! A few things to address

tycebrown committed:
fix: incorrect comment

tycebrown committed:
fix: show footnote to everyone

- footnote description is now only shown to language members.
- rename: canViewNote --> canReadLanguage

tycebrown committed:
consolidate paths

tycebrown committed:
update api docs

arrocke reviewed (approved):
Nice work!
